### PR TITLE
[SPARK-45982][INFRA] re-org R package installations

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
re-org R package installations

after this PR, there is no R package installation in `build_and_test`

### Why are the changes needed?
current R packages installations is somewhat messy, e.g. 
- `devtools` is installed 4 times (twice in dockerfile, twice in job `lint`)
- `testthat` is installed twice
- `roxygen2` is installed twice

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
@dongjoon-hyun @HyukjinKwon @Yikun

File: dev/infra/Dockerfile
Comment (dongjoon-hyun) [inline dev/infra/Dockerfile:75]: Or, shall you spin-off Python part because this PR title scope is `R package installation`, @zhengruifeng ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs).
Thank you for updating, @zhengruifeng .

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by zhengruifeng** [comment]
@dongjoon-hyun @HyukjinKwon thanks for the reviews!

**Comment by LuciferYang** [comment]
Will the R-related dependencies be installed in the daily tests of branch-3.x after this one merged?

**Comment by zhengruifeng** [comment]
@LuciferYang I am going to restore r installations in https://github.com/apache/spark/pull/43904

**Pull request closed without merging**
